tests/unit/amo/components/TestCategory.js

```diff
@@ -48,6 +72,11 @@ describe('Category', () => {
     sinon.assert.calledWith(fakeDispatch, categoriesFetch());
   });
 
+  it('renders 404 page for missing/invalid category', () => {
+    const root = renderHTML({ category: null });
```

Comment:
As mentioned you should be able to use enzyme for this test, though you might need to use `mount` or `render` because we're loading other components.